prow/spyglass/lenses/common/common.go

```diff
@@ -206,16 +204,16 @@ func FetchArtifacts(
 			_, err = art.Size()
 		}
 		if err != nil {
-			if name == "build-log.txt" {
-				podLogNeeded = true
+			if matched, err := regexp.MatchString(".*build-log.txt", name); err == nil && matched {
```

Comment:
The common way for such a static regex would be to define it as a package level variable like this, then you don't need the error checking and will discover during tests if it fails to compile:

`var buildLogRegex = regexp.MustCompile("regex string")`

Reply:
Also, not introduced by you, but why do we have both this and the `required_files` option? Do we ignore the `required_files` option or why is it possible to see the buildlog lense before there is anything in gcs?

Reply:
> Also, not introduced by you, but why do we have both this and the `required_files` option? Do we ignore the `required_files` option or why is it possible to see the buildlog lense before there is anything in gcs?

Yeah, so AFAICT we use this `ListArtifacts()` function in `cmd/deck/main.go` [here](https://github.com/kubernetes/test-infra/blob/e883d5a5273deafe70aa54754e1d4cb73f307c3a/prow/cmd/deck/main.go#L952) in order to list all the artifacts that either exist in gcs _or_ that we assume will exist in gcs in the future.  We use these artifact names to check against the `required_files` [here](https://github.com/kubernetes/test-infra/blob/e883d5a5273deafe70aa54754e1d4cb73f307c3a/prow/cmd/deck/main.go#L966).  We are able to see the buildlog lense even if there is no build-log in gcs because we explicitly add the buildlog artifact names in `ListArtifacts()`.  Before, this was done by adding a just a hardcoded `"build-log.txt"` and in this PR, we add buildlog artifact names by getting the container names.  Regardless, this logic is used so that we still render the buildlog lense even if there is not build-logs in gcs because we want still try to stream in the build-logs from the prowjob.